core/modules/widgets/widget.js

```diff
@@ -238,21 +238,64 @@ Widget.prototype.getStateQualifier = function(name) {
 /*
 Compute the current values of the attributes of the widget. Returns a hashmap of the names of the attributes that have changed
 */
+var AttributeTypes = null;
+
 Widget.prototype.computeAttributes = function() {
 	var changedAttributes = {},
 		self = this,
 		value;
-	$tw.utils.each(this.parseTreeNode.attributes,function(attribute,name) {
-		if(attribute.type === "filtered") {
-			value = self.wiki.filterTiddlers(attribute.filter,self)[0] || "";
-		} else if(attribute.type === "indirect") {
-			value = self.wiki.getTextReference(attribute.textReference,"",self.getVariable("currentTiddler"));
-		} else if(attribute.type === "macro") {
-			value = self.getVariable(attribute.value.name,{params: attribute.value.params});
-		} else { // String attribute
-			value = attribute.value;
+	if (!this.attributeGizmos) {
```

Comment:
`Gizmo` ... is there a more precise name?

Reply:
Haha, yeah, that's a half-formed thought.  Attribute value modules are very similar to widgets, but smaller in scope.  Widget and Gizmo are both terms for indefinite objects.  But I use the name `attributevalue` for the module-type.

I'd love some guidance on nomenclature.

Reply:
>Widget and Gizmo are both terms for indefinite objects.

I didn't know, that widget and gizmo actually have a very similar meaning. ... I should have consulted a "thesaurus" :)

>I'd love some guidance on nomenclature.

IMO it's ok to introduce new names, for new functionality. We just have to make sure, that they are consistent.